repository-ydb-v1/src/test/java/tech/ydb/yoj/repository/ydb/YdbRepositoryIntegrationTest.java

```diff
@@ -863,6 +866,30 @@ public void schemaWithHint() {
         repository.schema(HintAutoPartitioningByLoad.class).create();
     }
 
+
+    @Test
+    public void doNotCommitAfterTLI() {
+        Project.Id id1 = new Project.Id("id1");
+        Project.Id id2 = new Project.Id("id2");
+
+        RepositoryTransaction tx = repository.startTransaction(
+                TxOptions.create(IsolationLevel.SERIALIZABLE_READ_WRITE)
+                        .withImmediateWrites(true)
+                        .withFirstLevelCache(false)
+        );
+
+        tx.table(Project.class).find(id2);
+
+        db.tx(() -> db.projects().save(new Project(id2, "name2")));
+
+        tx.table(Project.class).save(new Project(id1, "name1")); // make tx available for TLI
+
+        assertThatExceptionOfType(OptimisticLockException.class)
+                .isThrownBy(() -> tx.table(Project.class).find(id2));
```

Comment:
Modern YDB versions can now return OptimisticLock from statements, it that correct?

Reply:
Yes, this test works over latest docker image (23.3).